### ISSUE TITLE
error_code=214 after first app load with asyncOpen

### ISSUE THREAD

**Issue by ritter1** [realm/realm-cocoa#6178]
## Goals
asyncOpen a realm in a freshly installed iOS app with RealmSwift 3.16.1

## Expected Results
Connect to and sync with realm in the app.

## Actual Results
```
ToDo[4923:23491947] Sync: Connection[1]: Session[1]: client_reset_config = true, Realm exists = false, async open = true, client reset = false
ToDo[4923:23491947] Sync: Connection[1]: Session[1]: Client reset config, metadata_dir = '/Users/xyz/Library/Developer/CoreSimulator/Devices/78F11C8E-D00C-46C5-98A0-B8EB0E104665/data/Containers/Data/Application/C58326A9-CE07-4305-8480-15EC5C24FC43/Documents/realm-object-server/io.realm.object-server-state/', recover_local_changes = true, require_recent_state_realm = true
ToDo[4923:23491947] Sync: Connection[1]: Connected to endpoint '3.122.70.141:443' (from '192.168.1.52:61900')
ToDo[4923:23491947] Sync: Connection[1]: Session[1]: Received: ERROR(error_code=214, message_size=21, try_again=0)
ToDo[4923:23491947] Sync: Connection[1]: Disconnected
```

## Steps to Reproduce
1. Setup https://github.com/realm/my-first-realm-app/tree/master/ios/01-query-based-sync with RealmSwift **3.13.1** and run in the simulator with iOS 12.2
2. After the first app start, create a nickname account within the app.
3. Delete the app from the simulator.
4. In WelcomeViewController.swift, line 66, comment out `initializeRealmPermissions(realm)` as this is supposed to run only once.
5. Install and run the app again on the simulator. > Runs fine!
6. Delete app again.
7. Upgrade RealmSwift to **3.16.1**. 
8. Install and run the app again on the simulator. > Received: ERROR(error_code=214, message_size=21, try_again=0)

## Code Sample
Code is used as in https://github.com/realm/my-first-realm-app/tree/master/ios/01-query-based-sync 

## Version of Realm and Tooling

Realm framework version: 3.16.1

Realm Object Server version: 3.23.1

Xcode version: 10.2.1 (10E1001)

iOS version: 12.2

**Comment by tgoyne** [comment]
asyncOpen doesn't do anything particularly interesting or useful on a query-based sync Realm, but it indeed shouldn't be outright failing like this.

**Comment by ritter1** [comment]
Thanks, received the fix in 3.16.2 just a couple of days after I reported the problem. Cool!
